src/language-js/utils.js

```diff
@@ -443,16 +443,28 @@ const flowTypeAnnotations = new Set([
   "BooleanLiteralTypeAnnotation",
   "StringTypeAnnotation",
 ]);
+const simpleTypeScriptTypeAnnotations = new Set([
+  "TSAnyKeyword",
+  "TSNullKeyword",
+  "TSTypeReference",
+  "TSThisType",
+  "TSNumberKeyword",
+  "TSVoidKeyword",
+  "TSBooleanKeyword",
+  "TSLiteralType",
```

Comment:
Please add more tests for `TSAnyKeyword`, `TSNullKeyword`, `TSThisType`, `TSVoidKeyword` and `TSLiteralType`.
I ran tests with the below code, but tests successful.
```ts
const simpleTypeScriptTypeAnnotations = new Set([
  // "TSAnyKeyword",
  // "TSNullKeyword",
  "TSTypeReference",
  // "TSThisType",
  "TSNumberKeyword",
  // "TSVoidKeyword",
  "TSBooleanKeyword",
  // "TSLiteralType",
]);
```

Reply:
I made this list according to `simpleFlowTypeAnnotations`.

Tests added https://github.com/prettier/prettier/pull/9521/commits/a51fb7f7eb417b55076ca66a2f451a338be3c41a